notebook2/notebook/notebook.py

```diff
@@ -175,8 +194,14 @@ def healthcheck():
     return '', 200
 
 
-@app.route('/')
+@app.route('/', methods=['GET'])
 def root():
+    return render_template('index.html')
```

Comment:
In this PR / and /notebook return an identical page (index.html). In the next PR, #5452, index.html becomes a proper home page, and the existing index.html is renamed notebook.html.

This is still secure, because the form present in index.html does not contain sensitive information, resources are not returned besides the html, and /new is protected.

Reply:
The existence of this comment suggests to me that these two PRs are not actually independent.

---

I'm a bit confused. The user is never redirected to `/notebook`, right? And there are no links to it? The endpoint seems to be never used.

---

It doesn't seem useful to permit a user to `GET /` where they'll fill out a form, which will get rejected (by `require_auth` on `new_post`). The rejection will redirect them to the login page, then they'll go to auth0 to log in, then they'll come back to notebook, and they'll get redirected to the "referrer", but the browser will probably try to GET that URL, which will send them to `GET /new` (not `POST /new`, for which their form data was lost already anyway), which will eventually redirect them back to `GET /` where they'll be, inexplicably, forced to fill the form again.

Seems to me we should just keep `/` and make it require auth.

Reply:
`/` becomes the homepage in the next PR. `/notebook` is notebook1's `/` 

GET `/new` will go away, I think it is not an appropriate way to handle clearing of resources.

Reply:
You can see that /new is gone here: https://github.com/hail-is/hail/pull/5476

These PRs are absolutely dependent. An independent PR, in my eyes, is a fully working notebook2. That would be ~500-1000 lines of code, which I thought was not the point.

At each PR prior to the last one, some feature is missing. Here, that feature is "home page", which the next PR addresses (https://github.com/hail-is/hail/pull/5452)

Reply:
If you prefer, I can fold the homepage PR into this one 
#5452.

Reply:
Ah, I realize now that I let the header go in with a broken link. I shouldn't have approved that.

Now that I've looked at the next PR, there's no need to reformulate this, I understand where we're going. For future PRs though, here's two thoughts:

In my view, an independent PR is one that a reviewer can confidently assess the correctness and usefulness of by looking at the PR's diff (combined with the reviewer's knowledge of the system).

The placeholder home page doesn't help me decide if we've correctly authenticated all the necessary endpoints, which I understand to be the main thrust of this PR. Moreover, this placeholder home page introduces the aforementioned bad flow. If users were actively using this system, I'd have to reject this PR because it would be immediately deployed and introduce this bad flow to our users.

Reply:
Sure, thanks for the feedback, I'll continue to refine. 

How would you handle a situation where the fully-working product takes ceil(N / 500 lines) PR's?